core/blockchain.go

```diff
@@ -1659,6 +1659,45 @@ const (
 	SideStatTy
 )
 
+// splitReceipts separates out the state-sync receipt from the whole receipt list
+// of a block and returns the encoded lists back separately. In case of errors or
+// empty receipt, it returns `nil` instead of `rlp.EncodeToBytes(nil)`.
+func splitReceipts(receipts rlp.RawValue, number uint64, hash common.Hash) (rlp.RawValue, rlp.RawValue) {
+	if receipts == nil {
+		return nil, nil
+	}
+
+	var decoded []*types.ReceiptForStorage
+	if err := rlp.DecodeBytes(receipts, &decoded); err != nil {
+		log.Warn("Failed to decode block receipts", "number", number, "hash", hash, "err", err)
+		return receipts, nil
+	}
+	// Find if there's a state-sync transaction receipt present. They are always
+	// appended at the end of list and can be identified by 0 gas usage.
+	if len(decoded) > 0 && decoded[len(decoded)-1].CumulativeGasUsed == 0 {
```

Comment:
Could you help me understand why `CumulativeGasUsed` is used to determine whether it is a state sync receipt instead of `GasUsed`?

Reply:
I am using GasUsed in Erigon. Let's agree on what's best here, thanks